scripts/biomedical/NCBI_Taxonomy/scripts/format_ncbi_taxonomy.py

```diff
@@ -0,0 +1,656 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Author: Pradeep Kumar Krishnaswamy
+Date: 18-Apr-2024
+Name: format_ncbi_taxonomy
+Description: cleaning the NCBI Taxonomy data.
+@source data: Download Taxdump.tar.z from NCBI Taxonomy FTP Download page
+@file_input: names.dmp, divisions.dmp, host.dmp & nodes.dmp
+@file_output: formatted ncbi_taxonomy.csv, tax_id_dcid_mapping.txt & ncbi_taxonomy_schema_enum.mcf
+"""
+
+# import the required packages
+import copy
+import re
+import string
+import sys
+import os
+from pathlib import Path
+import pandas as pd
+from absl import app
+from absl import flags
+from absl import logging
+
+SOURCE_FILE_PATH = None
```

Comment:
The SOURCE_FILE_PATH and OUTPUT_FILE_PATH should be pointing to the input/ and output/ subdirectories respectively. This currently does not run through from the head of the NCBI_Taxonomy folder as intended if you do the prescribed import procedure in the README.md file. Please fix the filepaths so that the scripts are stored and called from the scripts/ subdirectory, the source files are called from the input/ subdirectory, and the output files are written to the output/ subdirectory.

Reply:
The SOURCE_FILE_PATH and OUTPUT_FILE_PATH variables have been initialized with default value. These values will get updated either using absl flags or by passing arguments directly when running the script.